components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -370,6 +359,102 @@ public void testDelete(boolean isSudoing, boolean permDeleteOwned,
+      Project proj = mmFactory.simpleProject();
+      Dataset dat = mmFactory.simpleDataset();
+      Project sentProj = null;
+      Dataset sentDat = null;
+      sentProj = (Project) iUpdate.saveAndReturnObject(proj);
+      sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+      /* normalUser imports an image
+       * and targets it into the created Dataset.*/
+      List<IObject> originalFileAndImage = importImageWithOriginalFile(sentDat);
+      OriginalFile originalFile = (OriginalFile) originalFileAndImage.get(0);
+      Image image = (Image) originalFileAndImage.get(1);
+      assertOwnedBy(image, normalUser);
+      /* normalUser links the Project and the Dataset.*/
+      ProjectDatasetLink projectDatasetLink = linkParentToChild(sentProj, sentDat);
+      IObject datasetImageLink = iQuery.findByQuery(
+              "FROM DatasetImageLink WHERE child.id = :id",
+              new ParametersI().addId(image.getId()));
+      /* Post-import workflows are done either by lightAdmin or by otherUser.*/
+      if(isAdmin) {
+          loginUser(lightAdmin);
+      } else {
+          loginUser(otherUser);
+      }
+      client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
+      /* Check that lightAdmin or otherUser can delete the objects
+       * of normalUser only if lightAdmin has sufficient permissions or it is read-write group.
+       * Note that deletion of the Project
+       * would delete the whole hierarchy, which was successfully tested
+       * during writing of this test. The order of the below delete() commands
+       * is intentional, as the ability to delete the links and Project/Dataset/Image separately is
+       * tested in this way.
+       * Also check that the canDelete boolean
+       * on the object retrieved by the lightAdmin or otherUser matches the deletePassing
+       * boolean.*/
+      Assert.assertEquals(getCurrentPermissions(datasetImageLink).canDelete(), deletePassing);
+      doChange(client, factory, Requests.delete().target(datasetImageLink).build(), deletePassing);
+      Assert.assertEquals(getCurrentPermissions(projectDatasetLink).canDelete(), deletePassing);
+      doChange(client, factory, Requests.delete().target(projectDatasetLink).build(), deletePassing);
+      Assert.assertEquals(getCurrentPermissions(image).canDelete(), deletePassing);
+      doChange(client, factory, Requests.delete().target(image).build(), deletePassing);
+      Assert.assertEquals(getCurrentPermissions(sentDat).canDelete(), deletePassing);
+      doChange(client, factory, Requests.delete().target(sentDat).build(), deletePassing);
+      Assert.assertEquals(getCurrentPermissions(sentProj).canDelete(), deletePassing);
+      doChange(client, factory, Requests.delete().target(sentProj).build(), deletePassing);
+
+      /* Check the existence/non-existence of the objects as appropriate.*/
+      logRootIntoGroup(normalUser);
```

Comment:
Because the retrieval of objects in ``assertExists`` by the user who is logged in (``otherUser``) in private group fails -> the retrieval must thus be done by a more powerful user and root seemed a good choice.